pkg/k8s/pod.go

```diff
@@ -12,11 +14,27 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+const (
+	podConditionTypeForUnhealthy = v1.PodReady
+)
+
 var log = logf.Log.WithName("k8s.pod.go")
 
-// EvictPod evicts a single pod from Kubernetes
+// KillAndRemovePod immediately terminates a given pod off the node, without waiting to check if it has been removed.
+// This should only be called on workloads that can tolerate being removed like this, and as a last resort.
+func KillAndRemovePod(podName, podNamespace, nodeName string, client kubernetes.Interface) error {
```

Comment:
I think this should be `ForciblyDeletePod` - it better describes what we are actually performing on the pod.

Log lines below should be updated to match